TestExp1.test_branch_cut test failure with Intel Fortran

Running Windows Scipy 1.4.1 (compiled with MKL taken from https://www.lfd.uci.edu/~gohlke/pythonlibs/#scipy) standard test suite with Python 3.7.6 x64 results in failure for TestExp1.test_branch_cut test:
 
```
_________________________________________________________________________________________________________ TestExp1.test_branch_cut _________________________________________________________________________________________________________

self = <scipy.special.tests.test_exponential_integrals.TestExp1 object at 0x00000296E7953EC8>

    def test_branch_cut(self):
        assert np.isnan(sc.exp1(-1))
>       assert sc.exp1(complex(-1, 0)).imag == (
            -sc.exp1(complex(-1, -0.0)).imag
        )
E       AssertionError: assert -3.141592653589793 == --3.141592653589793
E        +  where -3.141592653589793 = (-1.8951178163559368-3.141592653589793j).imag
E        +    where (-1.8951178163559368-3.141592653589793j) = <ufunc 'exp1'>((-1+0j))
E        +      where <ufunc 'exp1'> = sc.exp1
E        +      and   (-1+0j) = complex(-1, 0)
E        +  and   -3.141592653589793 = (-1.8951178163559368-3.141592653589793j).imag
E        +    where (-1.8951178163559368-3.141592653589793j) = <ufunc 'exp1'>((-1-0j))
E        +      where <ufunc 'exp1'> = sc.exp1
E        +      and   (-1-0j) = complex(-1, -0.0)

self       = <scipy.special.tests.test_exponential_integrals.TestExp1 object at 0x00000296E7953EC8>

C:\Program Files\Python\3.7.6\lib\site-packages\scipy\special\tests\test_exponential_integrals.py:12: AssertionError
```

On the same PC package manually compiled with MKL on Ubuntu 18.04 (Python 3.6.5) successfully passes all tests.

I applied the patch to scipy 1.4.1 and rebuilt to no avail:
```
====================================================== FAILURES =======================================================
______________________________________________ TestExp1.test_branch_cut _______________________________________________

self = <scipy.special.tests.test_exponential_integrals.TestExp1 object at 0x000001C167954940>

    def test_branch_cut(self):
        assert np.isnan(sc.exp1(-1))
>       assert sc.exp1(complex(-1, 0)).imag == (
            -sc.exp1(complex(-1, -0.0)).imag
        )
E       AssertionError: assert -3.141592653589793 == --3.141592653589793
E        +  where -3.141592653589793 = (-1.8951178163559368-3.141592653589793j).imag
E        +    where (-1.8951178163559368-3.141592653589793j) = <ufunc 'exp1'>((-1+0j))
E        +      where <ufunc 'exp1'> = sc.exp1
E        +      and   (-1+0j) = complex(-1, 0)
E        +  and   -3.141592653589793 = (-1.8951178163559368-3.141592653589793j).imag
E        +    where (-1.8951178163559368-3.141592653589793j) = <ufunc 'exp1'>((-1-0j))
E        +      where <ufunc 'exp1'> = sc.exp1
E        +      and   (-1-0j) = complex(-1, -0.0)

D:\Programs\Python38\Lib\site-packages\scipy\special\tests\test_exponential_integrals.py:12: AssertionError
__________________________________________________ test_expi_complex __________________________________________________

    @check_version(mpmath, '0.10')
    def test_expi_complex():
        dataset = []
        for r in np.logspace(-99, 2, 10):
            for p in np.linspace(0, 2*np.pi, 30):
                z = r*np.exp(1j*p)
                dataset.append((z, complex(mpmath.ei(z))))
        dataset = np.array(dataset, dtype=np.complex_)

>       FuncData(sc.expi, dataset, 0, 1).check()
E       AssertionError:
E       Max |adiff|: 6.28319
E       Max |rdiff|: 0.304157
E       Bad results (9 out of 300) for the following points (in output 0):
E                           1.e-99+0.j => -227.37870854150898+6.283185307179586j !=        -227.37870854150898+0.j  (rdiff            0.02763312953742352)
E            1.668100537200083e-88+0.j => -201.5385869423536+6.283185307179586j !=        -201.53858694235356+0.j  (rdiff            0.03117609090400528)
E           2.7825594022071145e-77+0.j => -175.6984653431982+6.283185307179586j !=         -175.6984653431982+0.j  (rdiff            0.03576118490794107)
E           4.6415888336126776e-66+0.j => -149.8583437440428+6.283185307179586j !=         -149.8583437440428+0.j  (rdiff           0.041927497329819895)
E            7.742636826811214e-55+0.j => -124.01822214488739+6.283185307179586j !=        -124.01822214488739+0.j  (rdiff           0.050663404123299706)
E           1.2915496650148721e-43+0.j => -98.178100545732+6.283185307179586j !=           -98.178100545732+0.j  (rdiff            0.06399782917222806)
E           2.1544346900318602e-32+0.j => -72.3379789465766+6.283185307179586j !=          -72.3379789465766+0.j  (rdiff            0.08685873449436396)
E           3.5938136638045226e-21+0.j => -46.49785734742121+6.283185307179586j !=         -46.49785734742121+0.j  (rdiff            0.13512849119547773)
E            5.994842503189323e-10+0.j => -20.657735747666305+6.283185307179586j !=        -20.657735747666308+0.j  (rdiff            0.30415653409107984)

D:\Programs\Python38\Lib\site-packages\scipy\special\tests\test_mpmath.py:46: AssertionError
========================= 2 failed, 1353 passed, 164 skipped, 42 xfailed, 1 xpassed in 13.47s =========================
```

--- a/numpy/distutils/fcompiler/intel.py
+++ b/numpy/distutils/fcompiler/intel.py
@@ -59,7 +59,8 @@ class IntelFCompiler(BaseIntelFCompiler):
     def get_flags_opt(self):  # Scipy test failures with -O2
         v = self.get_version()
         mpopt = 'openmp' if v and v < '15' else 'qopenmp'
-        return ['-fp-model', 'strict', '-O1', '-assume', 'minus0', '-{}'.format(mpopt)]
+        return ['-fp-model', 'strict', '-O1',
+                '-assume', 'minus0', '-{}'.format(mpopt)]
 
     def get_flags_arch(self):
         return []

--- a/numpy/distutils/fcompiler/intel.py
+++ b/numpy/distutils/fcompiler/intel.py
@@ -59,7 +59,7 @@ class IntelFCompiler(BaseIntelFCompiler):
     def get_flags_opt(self):  # Scipy test failures with -O2
         v = self.get_version()
         mpopt = 'openmp' if v and v < '15' else 'qopenmp'
-        return ['-fp-model', 'strict', '-O1', '-{}'.format(mpopt)]
+        return ['-fp-model', 'strict', '-O1', '-assume', 'minus0', '-{}'.format(mpopt)]
 
     def get_flags_arch(self):
         return []
@@ -119,17 +119,6 @@ class IntelEM64TFCompiler(IntelFCompiler):
         'ranlib'       : ["ranlib"]
         }
 
-    def get_flags(self):
-        return ['-fPIC']
-
-    def get_flags_opt(self):  # Scipy test failures with -O2
-        v = self.get_version()
-        mpopt = 'openmp' if v and v < '15' else 'qopenmp'
-        return ['-fp-model', 'strict', '-O1', '-{}'.format(mpopt)]
-
-    def get_flags_arch(self):
-        return []
-
 # Is there no difference in the version string between the above compilers
 # and the Visual compilers?
 
@@ -174,7 +163,7 @@ class IntelVisualFCompiler(BaseIntelFCompiler):
         return ['/4Yb', '/d2']
 
     def get_flags_opt(self):
-        return ['/O1']  # Scipy test failures with /O2
+        return ['/O1', '/assume:minus0']  # Scipy test failures with /O2
 
     def get_flags_arch(self):
         return ["/arch:IA32", "/QaxSSE3"]

